pkg/log/zap/logger_test.go

```diff
@@ -152,7 +152,7 @@ func TestGetConfig(t *testing.T) {
 				sample: true,
 			},
 			expected: config{
-				encoder: jsonEncoder(),
+				encoder: newJSONEncoder(),
 				level:   zap.NewAtomicLevelAt(zapcore.Level(-10)),
 				opts:    append(opts, zap.AddStacktrace(zap.WarnLevel)),
 				sample:  false,
```

Comment:
Would you want to take a stab at adding some test cases that include the new `timeformatValue` stuff?

Reply:
I'm struggling quite a bit with getting the test framework to work on my local, but I can get some of the simpler (ie - non-integration) tests running. 

But it currently looks like the travis tests are failing although I can't quite figure out if it's related to a change I've made or not.

I see this error in the travis logs:
> memcached_test.go:114: timed out waiting for the condition
> (in this job: https://travis-ci.org/operator-framework/operator-sdk/jobs/545497878) 

I can't see how this is being caused by something I've done today, but I could easily be wrong about that. 

Are you able to help me resolve the current failing tests? I can then move on with confidence that the new tests I write should be good if I write them correctly. 

Finally, is it OK if I lean on the provided CI system to test tests as I develop them? Since I can't get most stuff running on my local, but I don't want to spam your CI system without permission :)

Reply:
I don't think these changes will affect the integraton/e2e tests that are failing in Travis. Those are somewhat flaky, so don't worry about the failures. There should be no need to run them locally as you're making changes, either.

The tests in this package are part of the unit tests that are run with `make test/unit`. Are you able to run that successfully?

Reply:
Thanks Joe - yes I can run `make test/unit` so that's been good. 

It doesn't look like there's any way I can inspect an EncoderConfig after the encoder has been initialised, so I can't figure out how to evaluate which time format the Encoder is using once it's been set. 

Without this functionality, the tests are pretty toothless and I'm not happy with them. 

I _think_ in order to really test if the time encoder is working we have to create a zap Entry and then parse the time format to see if it matches what zapcore should be using. I had a crack at this but wasn't able to make it work, but I think even if it didn't that's not a very good test because it's probably going to be too dependent on zaps' inner-workings rather than it's exposed API. 

Rather than pushing any further on that, I thought I'd send up what I have and see what you think? I'm running a bit low on time to keep working on this, but I'd really love to see it through to the end if you think there's an easy path to improving this test functionality?